### PR TITLE
chore: updating code owners file with design system team code ownership

### LINKED ISSUE #24313: Add design system team to codeowners
### What is this about?

Add https://github.com/orgs/MetaMask/teams/design-system-engineers to code owners file for ownership over `ui/components/component-library`

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Update .github/CODEOWNERS file 

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- Design system team is added as a reviewer for all changes to the ui/components/component-library folder
- Slack notifications is set up for design system engineers in the team channel

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This PR updates the CODEOWNERS file to reflect the new code ownership assignments for the `ui/components/component-library` folder, assigning ownership to the @MetaMask/design-system-engineers team. This change is based on the recent organizational updates documented in [this spreadsheet](https://docs.google.com/spreadsheets/d/1iSSQ9bTQ9apc7lzdxYheZ3P6eByH9E1y41WQx53036Q/edit#gid=0). The motivation behind this update is to ensure that the right team is notified for reviews on changes affecting the component library, thereby streamlining the review process and ensuring that changes align with our design system standards.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24313

## **Manual testing steps**

As this change pertains to GitHub repository settings and does not directly impact the application's functionality, manual testing steps are not applicable. However, the effectiveness of this change can be observed in future PRs affecting the `ui/components/component-library` folder, where the @MetaMask/design-system-engineers team should automatically be requested for review.

### **Before**

*Not applicable - This change does not affect the visual or functional aspects of the application.*

### **After**

*Not applicable - This change does not affect the visual or functional aspects of the application.*

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability.
- [x] I’ve included a link to the relevant documentation or spreadsheet justifying the change.
- [x] I’ve applied the right labels on the PR.

## **Pre-merge reviewer checklist**

- [x] I've reviewed the CODEOWNERS file update and confirmed it aligns with the documented organizational changes.
- [x] I confirm that this PR does not require manual testing and addresses the intended administrative change.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -59,4 +59,8 @@ ui/pages/confirmations               @MetaMask/confirmations-ux @MetaMask/confir
 ui/pages/institutional               @MetaMask/mmi
 ui/components/institutional          @MetaMask/mmi
 ui/ducks/institutional               @MetaMask/mmi
-ui/selectors/institutional           @MetaMask/mmi
\ No newline at end of file
+ui/selectors/institutional           @MetaMask/mmi
+
+# Design System to own code for the component-library folder
+# Slack handle: @metamask-design-system-team | Slack channel: #metamask-design-system
+ui/components/component-library      @MetaMask/design-system-engineers
```
Comment (georgewrmarshall) [inline .github/CODEOWNERS:66]: Seeing an error here for `Unknown owner` but I'm pretty sure its correct https://github.com/orgs/MetaMask/teams/design-system-engineers

Comment (garrettbear) [reply .github/CODEOWNERS:66]: Looks like it needs to be created

**Review by garrettbear — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (desi) [reply .github/CODEOWNERS:66]: It may not be added as a collaborator for this repo (the team itself) I will check with an admin.

Comment (georgewrmarshall) [reply .github/CODEOWNERS:66]: It has I think this is an issue with the linter? See handle here @MetaMask/design-system-engineers team

**Review by danjm — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (georgewrmarshall) [reply .github/CODEOWNERS:66]: This has been resolved by @danjm

**Pull request merged**